workspaces/diff-engine/src/events/mod.rs

```diff
@@ -43,7 +47,7 @@ impl Event for SpecEvent {
 }
 
 impl SpecEvent {
-  pub fn from_file(filename: &str) -> Result<Vec<SpecEvent>, EventLoadingError> {
+  pub fn from_file(filename: impl AsRef<Path>) -> Result<Vec<SpecEvent>, EventLoadingError> {
```

Comment:
Cool little bit I've learned: `impl AsRef<Path>` is used throughout `std::fs` and `tokio::fs` to express paths. Implemented from the most basic `&str` all the way up to `Path` and `PathBuf` itself.